packages/firestore/src/local/memory_remote_document_cache.ts

```diff
@@ -82,6 +82,19 @@ export class MemoryRemoteDocumentCache implements RemoteDocumentCache {
     return PersistencePromise.resolve(results);
   }
 
+  forEachDocumentKey(
+    transaction: PersistenceTransaction,
+    f: (key: DocumentKey) => PersistencePromise<void>
+  ): PersistencePromise<void> {
+    const iter = this.docs.getIterator();
```

Comment:
Is there a reason you are not just using `this.docs.forEach()`?

Reply:
Changed to `forEach`.